src/output.cpp

```diff
@@ -719,7 +719,7 @@ std::string string_replace( std::string text, const std::string &before, const s
 {
     // Check if there's something to replace (mandatory) and it's necessary (optional)
     // Second condition assumes that text is much longer than both &before and &after.
-    if( before.length() == 0 || !before.compare( after ) ) {
+    if( before.length() == 0 || before == after ) {
```

Comment:
```suggestion
    if( before.length() == 0 || before != after ) {
```

Reply:
Why? I interpreted this as:

return the original text, if string-to-remove is the same as the string to replace it with, as in:

`string_replace( some_text, "abc", "abc" )` will always yield `some_text` effectively unchanged.

Also: `!a.compare( b )` == `!( a.compare( b ) != 0 )` == `a.compare( b ) == 0` == `a == b`.

Reply:
Yes, you are correct.